lib/block-supports/typography.php

```diff
@@ -222,6 +222,200 @@ function gutenberg_typography_get_css_variable_inline_style( $attributes, $featu
+
+	// We get a 'preferred' unit to keep units consistent when calculating,
+	// otherwise the result will not be accurate.
+	$font_size_unit = isset( $minimum_font_size['unit'] ) ? $minimum_font_size['unit'] : 'rem';
+
+	// Grab the maximum font size and normalize it in order to use the value for calculations.
+	$maximum_font_size = gutenberg_get_typography_value_and_unit(
+		$maximum_font_size_raw,
+		array(
+			'coerce_to' => $font_size_unit,
+		)
+	);
+
+	// Protect against unsupported units.
+	if ( ! $maximum_font_size || ! $minimum_font_size ) {
+		return null;
+	}
+
+	// Use rem for accessible fluid target font scaling.
+	$minimum_font_size_rem = gutenberg_get_typography_value_and_unit(
+		$minimum_font_size_raw,
+		array(
+			'coerce_to' => 'rem',
+		)
+	);
+
+	// Viewport widths defined for fluid typography. Normalize units.
+	$maximum_viewport_width = gutenberg_get_typography_value_and_unit(
+		$maximum_viewport_width_raw,
+		array(
+			'coerce_to' => $font_size_unit,
+		)
+	);
+	$minimum_viewport_width = gutenberg_get_typography_value_and_unit(
+		$minimum_viewport_width_raw,
+		array(
+			'coerce_to' => $font_size_unit,
+		)
+	);
+
+	// Build CSS rule.
+	// Borrowed from https://websemantics.uk/tools/responsive-font-calculator/.
+	$view_port_width_offset = round( $minimum_viewport_width['value'] / 100, 3 ) . $font_size_unit;
+	$linear_factor          = 100 * ( ( $maximum_font_size['value'] - $minimum_font_size['value'] ) / ( $maximum_viewport_width['value'] - $minimum_viewport_width['value'] ) );
+	$linear_factor          = round( $linear_factor, 3 ) * $scale_factor;
+	$fluid_target_font_size = implode( '', $minimum_font_size_rem ) . " + ((1vw - $view_port_width_offset) * $linear_factor)";
+
+	return "clamp($minimum_font_size_raw, $fluid_target_font_size, $maximum_font_size_raw)";
+}
+
+/**
+ * Returns a font-size value based on a given font-size preset.
+ * Takes into account fluid typography parameters and attempts to return a css formula depending on available, valid values.
+ *
+ * @param array   $preset                      fontSizes preset value as seen in theme.json.
+ * @param boolean $should_use_fluid_typography An override to switch fluid typography "on". Can be used for unit testing.
+ * @return string Font-size value.
+ */
+function gutenberg_get_typography_font_size_value( $preset, $should_use_fluid_typography = false ) {
+	// Check if fluid font sizes are activated.
+	$typography_settings         = gutenberg_get_global_settings( array( 'typography' ) );
+	$should_use_fluid_typography = isset( $typography_settings['fluid'] ) && true === $typography_settings['fluid'] ? true : $should_use_fluid_typography;
+
+	if ( ! $should_use_fluid_typography ) {
+		return $preset['size'];
+	}
+
+	// Defaults.
+	$default_maximum_viewport_width   = '1600px';
+	$default_minimum_viewport_width   = '768px';
+	$default_minimum_font_size_factor = 0.75;
+	$default_maximum_font_size_factor = 1.5;
```

Comment:
I wonder if this should default to `1`. Then font sizes will stay as they are on larger screens, but rescale to be smaller on mobile.

Reply:
Alternatively, what if we set the scale to .5? My concern is that we don't want sizes to grow significantly on common viewport sizes.

Reply:
Alternatively, what if we set the scale to .5? My concern is that we don't want sizes to grow significantly on common viewport sizes.

Reply:
> I wonder if this should default to 1. Then font sizes will stay as they are on larger screens, but rescale to be smaller on mobile.

> Alternatively, what if we set the scale to .5? My concern is that we don't want sizes to grow significantly on common viewport sizes.

I agree, in the absence of any `fluid[min|max]` values, the `size` property should still _mean_ something when fluid typography is activated. 

For example, it could represent the max (or min font size) value as you suggest. Or it could be the middle point. There's probably an argument for the suitability of all three.

We'd have to be careful not to allow font sizes to become too small, especially on mobile, but also not too large.

In general, I think it's something we can play with to achieve a balanced spread. 

I'll land this and we can test further. Thanks a lot!